### PR TITLE
[SELS-TASK] [FE] Create Add Word page

### PR DESCRIPTION
**Pull request by wylenjoan**
#### [Asana Link]
https://app.asana.com/0/1202959865559554/1202970748735099

#### [Commands]
**Backend**

Activate the virtual environment:
`cd backend`
`venv/Scripts/Activate`

Run backend:
`py manage.py runserver`

**Frontend**
Run frontend:
`cd frontend`
`npm run start`

#### [Pre-conditions]
- The reviewer must test on http://127.0.0.1:3000/
- The reviewer must log in as a user with admin role.

#### [Expected Output]
- Should be able to add word with choices in a category
- Should be able to view all words with choices by category

#### [Notes]
- The page to add words with choices is http://127.0.0.1:3000/category-list

127 0 0 1_3000_category-list

> Word list
127 0 0 1_3000_category-list (1)

### CODE REVIEW

File: backend/app/views.py [hunk 1]
```diff
@@ -94,6 +94,17 @@ class GetUpdateDeleteQuestion(generics.RetrieveUpdateDestroyAPIView):
   serializer_class = QuestionSerializer
 
 
+class ListQuestionByCategory(generics.ListAPIView):
+  serializer_class = QuestionSerializer
+
+  def get_queryset(self):
+    queryset = Question.objects.all()
+    category = self.request.query_params.get('category')
+    if category is not None:
```
Comment (ziontavera) [inline backend/app/views.py]:
```suggestion
    if category:
```

File: backend/app/views.py [hunk 2]
```diff
@@ -94,6 +94,17 @@ class GetUpdateDeleteQuestion(generics.RetrieveUpdateDestroyAPIView):
   serializer_class = QuestionSerializer
 
 
+class ListQuestionByCategory(generics.ListAPIView):
+  serializer_class = QuestionSerializer
+
+  def get_queryset(self):
```
Comment (ziontavera) [inline backend/app/views.py:100]: you can actually put `request` beside `self`
```
  def get_queryset(self, request):
```
para `request.query_params` nalang siya largo. No need to change your code. Just a note lang

Other stuff nga pwede sad is:

```
def func(self, request, *args, **kwargs):
```

Depends on how you passed the data from FE

File: backend/app/views.py (hunk 1 above)
Comment (ziontavera) [inline backend/app/views.py]:
```suggestion
    if category:
```

File: frontend/src/components/AddWordModal.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    handleClose,
+    handleCreateQuestionWithChoices,
+  } = props;
+
+  const [word, setWord] = useState<Question>({ value: "" });
+  const [choices, setChoices] = useState<Choice[]>([
+    { value: "", is_correct_answer: false },
+    { value: "", is_correct_answer: false },
+    { value: "", is_correct_answer: false },
+    { value: "", is_correct_answer: false },
```
Comment (ziontavera) [inline frontend/src/components/AddWordModal.tsx:26]: can this be dynamic? Depending on what is returned from BE

Comment (wylenjoan) [reply frontend/src/components/AddWordModal.tsx:26]: gituyo nako ni sir para ani:
image

or unsay better way to implement this sir? hehe

Comment (ziontavera) [reply frontend/src/components/AddWordModal.tsx:26]: @wylenjoan Hmmm its kinda static. Akong concern lang is what if mag add ug Choice? Mag add sad ka ug extra object sa state? Maka add ba ug choice? or 4 ra kutob?

Comment (wylenjoan) [reply frontend/src/components/AddWordModal.tsx:26]: @ziontavera Wala ko nakathink nga pwede mag-add ug choice sir hahaha. Maybe static lang sah for now unya will improve it if naay extra time 😅

**Review by ziontavera — APPROVED** [review]

**Pull request merged**